src/adapter/src/client.rs

```diff
@@ -218,6 +227,28 @@ impl Drop for ConnClient {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ExtendedRequest {
+    #[serde(alias = "sql")]
+    /// 0 or 1 queries
+    pub query: String,
+    /// Optional parameters for the query
+    pub params: Option<Vec<Option<String>>>,
```

Comment:
Is `Vec<Option<String>>` ok here instead? Empty vec is reasonable for no params.

Reply:
That would require the `params` key; seems annoying to have to specify an empty array and is a nominal amount of complexity on our end.

Reply:
I think you could stick a `#[serde(default)]` attribute on the field, and then serde would produce the empty vec if the field was missing from the user request.